samples/jpegDecodePerf/jpegdecodeperf.cpp

```diff
@@ -0,0 +1,353 @@
+                valid_rocjpeg_stream_handles = rocjpeg_stream_handles;
+                valid_subsamplings = subsamplings;
+                valid_widths = widths;
+                valid_heights = heights;
+                valid_prior_channel_sizes = prior_channel_sizes;
+                valid_output_images = output_images;
+                valid_base_file_names = base_file_names;
+            }
+        }
+
+        double time_per_batch_in_milli_sec = 0;
+        if (current_batch_size > 0) {
+            auto start_time = std::chrono::high_resolution_clock::now();
+            CHECK_ROCJPEG(rocJpegDecodeBatched(rocjpeg_handle, valid_rocjpeg_stream_handles.data(), current_batch_size, &decode_params, valid_output_images.data()));
+            auto end_time = std::chrono::high_resolution_clock::now();
+            time_per_batch_in_milli_sec = std::chrono::duration<double, std::milli>(end_time - start_time).count();
+        }
+
+        double image_size_in_mpixels = 0;
+        for (int b = 0; b < current_batch_size; b++) {
+            image_size_in_mpixels += (static_cast<double>(valid_widths[b][0]) * static_cast<double>(valid_heights[b][0]) / 1000000);
+        }
+
+        *num_decoded_images += current_batch_size;
+
+        if (save_images) {
+            for (int b = 0; b < current_batch_size; b++) {
+                std::string image_save_path = output_file_path;
+                //if ROI is present, need to pass roi_width and roi_height
+                uint32_t width = is_roi_valid ? roi_width : valid_widths[b][0];
+                uint32_t height = is_roi_valid ? roi_height : valid_heights[b][0];
+                rocjpeg_utils.GetOutputFileExt(decode_params.output_format, valid_base_file_names[b], width, height, valid_subsamplings[b], image_save_path);
+                rocjpeg_utils.SaveImage(image_save_path, &valid_output_images[b], width, height, valid_subsamplings[b], decode_params.output_format);
+            }
+        }
+
+        total_decode_time_in_milli_sec += time_per_batch_in_milli_sec;
+        image_size_in_mpixels_all += image_size_in_mpixels;
+
+        bad_image_indices.clear();
+    }
+
+    double avg_time_per_image = total_decode_time_in_milli_sec / *num_decoded_images;
+    *images_per_sec = 1000 / avg_time_per_image;
+    *image_size_in_mpixels_per_sec_all = *images_per_sec * (image_size_in_mpixels_all / *num_decoded_images);
+
+    for (auto& it : output_images) {
+        for (int i = 0; i < ROCJPEG_MAX_COMPONENT; i++) {
+            if (it.channel[i] != nullptr) {
+                CHECK_HIP(hipFree((void *)it.channel[i]));
+                it.channel[i] = nullptr;
+            }
+        }
+    }
+}
+
+int main(int argc, char **argv) {
+    int device_id = 0;
+    bool save_images = false;
+    int num_threads = 1;
+    int batch_size = 1;
+    bool is_dir = false;
+    bool is_file = false;
+    std::string input_path, output_file_path;
+    std::vector<std::string> file_paths = {};
```

Comment:
suggest putting all information needed for each thread into a single struct like DecodeInfo like below
DecodeInfo {
    std::vector<std::string> jpeg_files;
    ...
}
and allocate number of threads instances of the struct for each thread. Then pass a reference to struct instance for each thread function

Reply:
Done.